### PR TITLE
Feat/condo/doma 7026/two billing accounts no service consumer

### PR DESCRIPTION
**Pull request by AleX83Xpert**

### CODE REVIEW

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js [hunk 1]
```diff
@@ -121,17 +133,18 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
...
                 await loadListByChunks({
                     context,
                     list: Property,
                     chunkSize: 50,
                     where: {
                         deletedAt: null,
-                        OR: items.map((item) => ({
+                        OR: billingAccountItemsData.map((item) => ({
                             AND: [
                                 { organization: { id: item.organizationId } },
```
Comment (AntonAL) [inline apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:149]: Maybe also to include `deletedAt`?

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js [hunk 2]
```diff
@@ -146,37 +159,95 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
...
+                // with the same period
+                /** @type {Object<string, Set>} */
+                const billingAccountsByAddressAndCategoryAndPeriod = {}
+                for (const item of billingAccountItemsData) {
+                    // We decided to check 2 last receipts
+                    /** @type {BillingReceipt[]} */
+                    const receipts = await BillingReceipt.getAll(context, {
+                        deletedAt: null,
+                        context: {
+                            organization: { id: item.organizationId },
```
Comment (AntonAL) [inline apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:175]: Maybe to include `deletedAt: null`?

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js [hunk 3]
```diff
@@ -146,37 +159,95 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
...
+                        },
+                        account: {
+                            OR: [
+                                { property: { address: item.address } },
```
Comment (AntonAL) [inline apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:179]: Maybe to include `deletedAt: null`?

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js [hunk 4]
```diff
@@ -146,37 +159,95 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
...
+                                { property: { addressKey: item.addressKey } },
+                            ],
+                            unitType: item.unitType,
+                            unitName: item.unitName,
+                        },
+                    }, { sortBy: ['period_DESC'], first: 2 })
+
+                    for (const receipt of receipts) {
+                        const theKey = `${item.address}_${item.unitType}_${item.unitName}_${receipt.category.id}_${receipt.period}`
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:188]: it's better to use hash here

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js [hunk 5]
```diff
@@ -102,7 +114,7 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
                     chunkSize: 50,
                     where: {
                         deletedAt: null,
-                        organization: { id_in: items.map(({ organizationId }) => organizationId) },
+                        organization: { id_in: billingAccountItemsData.map(({ organizationId }) => organizationId) },
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:117]: done

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:149]: yes, done

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js (hunk 3 above)
Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:179]: ok

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js (hunk 4 above)
Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js]: done

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js [hunk 6]
```diff
@@ -64,36 +64,48 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
...
+                                billingAccountItemsData.push({ id, organizationId, address, addressKey, unitType, unitName, number, billingContextId })
+                            }
+                        }
+
+                        return []
+                    },
+                })
 
                 // Filter by acquiring context
+                // The organization must have the finished acquiring context
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js:112]: Not quite accurate
image
Looks like the first variant is better
image

File: apps/condo/domains/resident/schema/DiscoverServiceConsumersService.test.js [hunk 7]
```diff
@@ -36,7 +36,7 @@ const {
     makeClientWithProperty, createTestProperty, updateTestProperty,
 } = require('@condo/domains/property/utils/testSchema')
 const { buildFakeAddressAndMeta } = require('@condo/domains/property/utils/testSchema/factories')
-const { MAX_RESIDENT_DISCOVER_CONSUMERS_BY_WINDOW_SEC } = require('@condo/domains/resident/constants/constants')
+const { MAX_RESIDENT_DISCOVER_CONSUMERS_BY_WINDOW_SEC, PAYMENT_CATEGORIES_META } = require('@condo/domains/resident/constants/constants')
```
Comment (dkoviazin) [inline apps/condo/domains/resident/schema/DiscoverServiceConsumersService.test.js:39]: PAYMENT_CATEGORIES_META needs to die

**Review by dkoviazin — APPROVED** [review]

**Pull request merged**